src/EnergyPlus/SetPointManager.cc

```diff
@@ -8429,6 +8459,43 @@ namespace SetPointManager {
         }
     }
 
+    int getSPMBasedOnNode(int const NodeNum, int const SetPtType, int const SPMType, CtrlNodeType ctrlOrRefNode) {
+
+        if (GetInputFlag) {
+            GetSetPointManagerInputs();
+            GetInputFlag = false;
+        }
+
+        int getSPMBasedOnNode = 0;
+
+        for (int SetPtMgrNum = 1; SetPtMgrNum <= NumAllSetPtMgrs; ++SetPtMgrNum) {
+            if (SetPtType == AllSetPtMgr(SetPtMgrNum).CtrlTypeMode) { // SetPtType is e.g., iCtrlVarType_Temp, iCtrlVarType_HumRat, etc.
+                switch (ctrlOrRefNode) { // ctrlOrRefNode is enum type of node to look for, either control node or reference node
+                case CtrlNodeType::control: {
+                        for (int NumNode = 1; NumNode <= AllSetPtMgr(SetPtMgrNum).NumCtrlNodes; ++NumNode) {
+                            // SPMType is type of set point manager, e.g., iSPMType_Scheduled, iSPMType_MixedAir, etc.
+                            if (NodeNum == AllSetPtMgr(SetPtMgrNum).CtrlNodes(NumNode) && SPMType == AllSetPtMgr(SetPtMgrNum).SPMType) {
+                                getSPMBasedOnNode = AllSetPtMgr(SetPtMgrNum).SPMIndex; // SPMIndex is index to specific type of SPM
+                                break;
+                            }
+                        }
+                        break;
+                }
+                case CtrlNodeType::reference: {
+                        // SPMType is type of set point manager, e.g., iSPMType_Scheduled, iSPMType_MixedAir, etc.
+                        if (NodeNum == AllSetPtMgr(SetPtMgrNum).RefNode && SPMType == AllSetPtMgr(SetPtMgrNum).SPMType) {
+                            getSPMBasedOnNode = AllSetPtMgr(SetPtMgrNum).SPMIndex; // SPMIndex is index to specific type of SPM
+                        }
+                        break;
+                }
+                }
+            }
+            if (getSPMBasedOnNode > 0) break;
```

Comment:
I wondered if I should break here or continue to find a duplicate if a user puts 2 of the same SPM on the same node. Since it's not a fatal if a user puts 2 SPMs on the same node (a warning is currently issued) then no need to continue (at this point in time, maybe needed later)

Reply:
Seems reasonable for now and potentially changing later.